### PR TITLE
[dom-gpu, daint-gpu] Eb templates in boost

### PR DESCRIPTION
**Pull request by ekouts**
Since #1616 is getting too big, I am splitting the PR into smaller ones.

### CODE REVIEW

File: easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python2.eb [hunk 1]
```diff
@@ -45,7 +40,7 @@ dependencies = [
 # for python3 the corresponding lib is libboost_python36.so
 sanity_check_paths = {
     'files': ['lib/libboost_system.so', 'lib/libboost_mpi.so',
-              'lib/libboost_python%s%s.so' % (local_py_maj_ver, local_py_min_ver)],
+              'lib/libboost_python%%(pymajver)s%s.so' % _pyminver],
```
Comment (lucamar) [inline easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python2.eb:43]: Thanks @ekouts: is it possible to use the same format for both `major` and `minor` versions, as in my suggestion ? 
```suggestion
              'lib/libboost_python%s%s.so' % (pymajver, _pyminver)],
```

File: easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python3.eb [hunk 2]
```diff
@@ -46,7 +41,7 @@ patches = ['Boost-1.70.0-python%s.patch' % local_py_maj_ver]
 # for python3 the corresponding lib is libboost_python36.so
 sanity_check_paths = {
     'files': ['lib/libboost_system.so', 'lib/libboost_mpi.so',
-              'lib/libboost_python%s%s.so' % (local_py_maj_ver, local_py_min_ver)],
+              'lib/libboost_python%%(pymajver)s%s.so' % _pyminver],
```
Comment (lucamar) [inline easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python3.eb:44]: Thanks @ekouts: same comment here as above for `majver` and `minver`, just to have the same formatting, if possible

File: easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python2.eb (hunk 1 above)
Comment (ekouts) [reply easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python2.eb:43]: `pymajver` is a template variable, while `_pyminver` is a local variable. The problem is template variables don't work with this syntax.

Comment (lucamar) [reply easybuild/easyconfigs/b/Boost/Boost-1.70.0-CrayGNU-19.10-python2.eb:43]: Ok, never mind!

**Pull request merged**